Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -742,34 +745,34 @@ namespace Babylon
         const std::string fragmentSource = info[1].As<Napi::String>().Utf8Value();
         const Napi::Function onSuccess = info[2].As<Napi::Function>();
         const Napi::Function onError = info[3].As<Napi::Function>();
+
         ProgramData* program = new ProgramData{};
+        Napi::Value jsProgram = Napi::Pointer<ProgramData>::Create(info.Env(), program, Napi::NapiPointerDeleter(program));
+
         arcana::make_task(arcana::threadpool_scheduler, *m_cancellationSource,
-            [this, vertexSource, fragmentSource, program, cancellationSource{m_cancellationSource}]() -> std::pair<bgfx::ProgramHandle, std::unordered_map<std::string, uint32_t>>
+            [this, vertexSource, fragmentSource, program, cancellationSource{m_cancellationSource}]() -> std::unique_ptr<ProgramData>
             {
-                return CreateProgramInternal(vertexSource, fragmentSource, *program);
+                return CreateProgramInternal(vertexSource, fragmentSource);
             })
-            .then(m_runtimeScheduler, *m_cancellationSource, [program, onSuccessRef{Napi::Persistent(onSuccess)}, onErrorRef{Napi::Persistent(onError)}, cancellationSource{m_cancellationSource}](arcana::expected<std::pair<bgfx::ProgramHandle, std::unordered_map<std::string, uint32_t>>, std::exception_ptr> result)
+            .then(m_runtimeScheduler, *m_cancellationSource,
+                [program,
+                jsProgramRef{Napi::Persistent(jsProgram)},
+                onSuccessRef{Napi::Persistent(onSuccess)},
+                onErrorRef{Napi::Persistent(onError)},
+                cancellationSource{m_cancellationSource}](const arcana::expected<std::unique_ptr<ProgramData>, std::exception_ptr>& result)
```

Comment:
We use clang-format. VS should use the `.clang-format` file at the root of the repo if configured properly. Please format accordingly.

Reply:
Make sure your VS formatting settings are set to use clang-format.

<img width="139" alt="image" src="https://user-images.githubusercontent.com/19158740/223790302-c8999d06-2879-464e-afe2-7e6e8091eee5.png">

Reply:
It was already turned on. For whatever reason it wasn't changing this code on save. No idea why, but I hope it's the way you want it now.